File: aat-android/src/main/java/ch/bailu/aat/views/preferences/SolidDirectoryViewSAF.java
package ch.bailu.aat.views.preferences;

import android.app.Activity;

import ch.bailu.aat.util.ui.theme.UiTheme;
import ch.bailu.aat_lib.preferences.SolidFile;

public class SolidDirectoryViewSAF extends SolidDirectoryView {
    private final Activity acontext;

    public SolidDirectoryViewSAF(Activity ac, SolidFile s, UiTheme theme) {
        super(ac,s, theme);
        acontext = ac;

    }

    @Override
    public void onRequestNewValue() {
        new SolidDirectoryDialog(acontext, solid);
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/views/preferences/SolidDirectoryViewSAF.kt
package ch.bailu.aat.views.preferences

import android.app.Activity
import ch.bailu.aat.util.ui.theme.UiTheme
import ch.bailu.aat.views.preferences.dialog.SolidDirectoryDialog
import ch.bailu.aat_lib.preferences.SolidFile

class SolidDirectoryViewSAF(private val acontext: Activity,
                            private val solidFile: SolidFile,
                            theme: UiTheme) :
    AbsSolidView(acontext, solidFile, theme) {
    override fun onRequestNewValue() {
        SolidDirectoryDialog(acontext, solidFile)
    }
}
